encoder/libav_encoder.cpp

```diff
@@ -388,7 +405,13 @@ void LibAvEncoder::initOutput()
 	if (!(out_fmt_ctx_->flags & AVFMT_NOFILE))
 	{
 		std::string filename = options_->output;
-
+		if (options_->segment){			
+			char subfilename[256];
+			snprintf(subfilename, sizeof(subfilename), options_->output.c_str(), segment_num);
```

Comment:
I have a bad feeling this might crash if `options_->output.c_str()` is too long?

Reply:
Ah no, re-reading the snprintf documentation this does indeed seem to be oK!!